context.go

```diff
@@ -6,53 +6,142 @@ import (
 	"flag"
 	"fmt"
 	"strings"
+	"time"
 )
 
-// Context is a type that is passed through to
-// each Handler action in a cli application. Context
+// Context is an interface that is passed through to
+// each Handler action in a cli application. It
 // can be used to retrieve context-specific args and
 // parsed command-line options.
-type Context struct {
-	context.Context
-	App           *App
-	Command       *Command
+type Context interface {
+	// Functions to be able to manage Context type
+	App() *App
+	SetCommand(command *Command)
```

Comment:
It looks like this is only used immediately after creating a new `cli.Context` for the command. Since this is something users won't ever be calling directly, we should see if we can't remove this from the public interface.